apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionSteps/TransactionSteps.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+
+import { ethers } from 'ethers';
+import { parseUnits } from 'ethers/lib/utils';
+
+import { Button, Icon, IconNames, StatusType } from '@sovryn/ui';
+
+import { Transaction, TxConfig } from '../../TransactionStepDialog.types';
+import { TransactionStep } from '../TransactionStep/TransactionStep';
+
+export type TransactionStepsProps = {
+  transactions: Transaction[];
+  onSuccess?: () => void;
+  gasPrice: string;
+};
+
+export const TransactionSteps: FC<TransactionStepsProps> = ({
+  transactions,
+  onSuccess,
+  gasPrice,
+}) => {
+  const [configs, setConfigs] = useState<TxConfig[]>([]);
+  const [step, setStep] = useState(-1);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    const initilize = async () => {
+      const list: TxConfig[] = [];
+      for (let i = 0; i < transactions.length; i++) {
+        const tx = transactions[i];
+        const gasLimit = await tx.contract.estimateGas[tx.fnName](...tx.args);
+        list.push({
+          ...tx.config,
+          amount: tx.fnName === 'approve' ? tx.args[1] : undefined,
+          unlimitedAmount: tx.fnName === 'approve' ? false : undefined,
+          gasPrice,
+          gasLimit: gasLimit.toString(),
+        });
+      }
+      setConfigs(list);
+    };
+
+    if (gasPrice) {
+      initilize();
+    }
+  }, [gasPrice, transactions]);
+
+  const updateConfig = useCallback(
+    (index: number, config: TxConfig) => {
+      const list = [...configs];
+      list[index] = { ...config };
+      setConfigs(list);
+    },
+    [configs],
+  );
+
+  const submit = useCallback(async () => {
+    try {
+      let i = 0;
+      if (error) {
+        setError(false);
+        i = step;
+      }
+      for (; i < transactions.length; i++) {
+        setStep(i);
+        const config = configs[i];
+        const args = transactions[i].args;
+        if (transactions[i].fnName === 'approve') {
+          args[1] = config.unlimitedAmount
+            ? ethers.constants.MaxUint256
+            : config.amount;
+        }
+        const tx = await transactions[i].contract[transactions[i].fnName](
+          ...args,
+          {
+            ...transactions[i].config,
+            gasPrice: config.gasPrice
+              ? parseUnits(config.gasPrice?.toString() || '0', 9)
+              : undefined,
+            gasLimit: config.gasLimit ? config.gasLimit?.toString() : undefined,
+          },
+        );
+
+        configs[i] = { ...config, hash: tx.hash };
+        setConfigs([...configs]);
+
+        await tx.wait();
+        transactions[i].onComplete?.(tx.hash);
+      }
```

Comment:
Instead of looping through transactions and confirming each of them by a single button submission, I think we should ask user to submit button for each of transactions (ofc only once the transaction before it was confirmed)

Reply:
Well the logic in the spec is like that. I think it's easier for the user to have it automatic.
@soulBit what do you think?

Reply:
It can be like this at this point, but having it without user action may crash dapp for HW devices or even metamask when working for networks that can confirm transactions instantly (bsc for example)

Reply:
Victor and I discussed it and he convinced me this is a good idea, but I want to get Product sign off before we implement - let's leave as-is for now and update it later.